docs/features/pixi_global_manifest.md

```diff
@@ -0,0 +1,222 @@
+# Pixi Global Manifest
+
+## Motivation
+
+`pixi global` is currently limited to imperatively managing CLI packages.
+The next iteration of this feature should fulfill the following needs:
+
+- Shareable global environments.
+- Managing complex environments with multiple packages as dependencies
+- Flexible exposure of binaries
+
+## Design Considerations
+
+There are a few things we wanted to keep in mind in the design:
+
+1. **User-friendliness**: Pixi is a user focused tool that goes beyond developers. The feature should have good error reporting and helpful documentation from the start.
+2. **Keep it simple**: The CLI should be all you strictly need to interact with global environments.
+3. **Unsurprising**: Simple commands should behave similar to traditional package managers.
+4. **Human Readable**: Any file created by this feature should be human-readable and modifiable.
+
+## Manifest
+
+```toml title="pixi-global.toml"
+# The name of the environment is `python`
+# It will expose python, python3 and python3.11, but not pip
+[envs.python.dependencies]
+python = { spec = "3.11.*", expose_binaries = "auto" }
+pip = "*"
+
+# The name of the environment is `python_3_10`
+# It will expose python3.10
+[envs.python_3_10.dependencies]
+python = { spec = "3.10.*", expose_binaries = { "python3.10"="python" } }
+
+```
+
+## CLI
+
+Install one or more packages `PACKAGE` into their own environments and expose their binaries.
+If no environment named `PACKAGE` exists, it will be created.
+The syntax for `MAPPING` is `exposed_name=binary_name`, so for example `python3.10=python`.
+
+```
+pixi global install [--expose MAPPING] <PACKAGE>...
+```
+
+Remove environments `ENV`.
+```
+pixi global uninstall [ENV]...
+```
+
+Upgrade all packages in environments `ENV`
+```
+pixi global upgrade [ENV]...
+```
+
+Inject package `PACKAGE` into an existing environment `ENV`.
+If environment `ENV` does not exist, it will return with an error.
+```
+pixi global inject --environment ENV PACKAGE
+```
+
+Remove package `PACKAGE` from environment `ENV`.
+If that was the last package remove the whole environment and print that information in the console.
+```
+pixi global remove --environment ENV PACKAGE
```

Comment:
I agree, we discussed that too.

There are multiple options I see:

`pixi global add` and `pixi global remove`: one criticism was that it isn't easy to guess how `add` and `install` differs

`pixi global inject` and `pixi global uninject`: `uninject` could also remove packages that were not injected but installed. Might be a bit surprising, I think

`pixi global inject` and `pixi global remove`: `remove` is not the opposite of `inject`, which I think is fine? `remove` can remove any packages in the environment

Reply:
Also `pipx` has extra metadata for what packages it injected, so you cannot uninject what was it has not injected. I propose you do something similar and go for `inject` `uninject`. Or is that not possible?

Reply:
We could make it so that `uninject` only works for packages that don't set the `expose_binaries` key.

Does that make sense to you.
Do we then still need a `pixi global remove` command that can remove any packages?

Reply:
I dont see the point of being able to remove "any" package. I assume that both `install` and `inject` add "top-level" dependencies, e.g. dependencies that the user directly requests when solving the environment. This results in a minimal compatible solution. Removing any other (thus transitive) dependency would break the environment. Or am I missing something.

Reply:
> Removing any other (thus transitive) dependency would break the environment. Or am I missing something.

This is a good point to bring up.
I wasn't talking about transitive dependencies, but about dependencies that expose binaries.

Let's take this manifest for example:

```toml
[envs.python.dependencies]
python = { spec = "3.11.*", expose_binaries = "auto" }
bat = { spec = "*", expose_binaries = "auto" }
pip = "*"
```

If I understood @tdejager correctly, `pip uninject pip` would work, but `pip uninject bat` would not.